lib/exhal/form.ex

```diff
@@ -0,0 +1,140 @@
+defmodule ExHal.Form do
+  @moduledoc """
+
+  Represents a [Dwolla style HAL
+  form](https://github.com/Dwolla/hal-forms). Generally these are
+  acquired from `ExHal.Document.get_form/2`,
+  `ExHal.Document.fetch_form/2`, etc
+
+  """
+
+  alias ExHal.{
+    FormField,
+    JsonFormEncoder
+  }
+
+  @typedoc """
+  A from that can be completed and submitted. This type is opaque and
+  should use only as a argument to functions in this `ExHal.Form` module.
+  """
+  @opaque t :: %__MODULE__{}
+
+  defstruct [
+    :target,
+    :method,
+    :content_type,
+    :fields
+  ]
+
+  @spec from_forms_entry(%{}) :: __MODULE__.t()
```

Comment:
I'm surprised to see your `@spec`s above your `@doc`s. Is this the new standard from the formatter?

Reply:
i don't think the formatter has an opinion on this. i don't have an opinion on this. do you, and others, prefer docs before specs?

Reply:
I do, because that's what I'm more used to seeing, but I don't have a strong opinion. FWIW, docs before specs is how the core of Elixir is written.

Reply:
Also FWIW, I notice in the other PR that the docs are before specs in `client.ex`.